src/annotator/anchoring/text-range.ts

```diff
@@ -149,7 +149,14 @@ export class TextPosition {
     offset: number;
   } {
     try {
-      return resolveOffsets(this.element, this.offset)[0];
+      const results = resolveOffsets(this.element, this.offset)[0];
+      if (
+        options.direction === ResolveDirection.BACKWARDS &&
+        results.offset === 0
+      ) {
+        throw new RangeError('Text node at end of range has no selected text');
```

Comment:
The error message here refers to "range", which may be inappropriate. I can tune later if we move forward.